### ISSUE TITLE
Authentication frontend tasks

### ISSUE THREAD

**Issue by madrix01** [AOSP-NIT-Surat/poller#2]
# Frontend Tasks

Try to keep UI as much as minimal no fancy colors or anything [ just a suggestion ]
- [ ] Make page for login
- [ ] Make page for registration 

> While making PR or issue for frontend always include screenshot with it. 
Thank you for contributing !

**Comment by Harshil-Jani** [comment]
Hey! Can I try for this ?

**Comment by madrix01** [comment]
@Harshil-Jani sure!!!

**Comment by prakhar728** [comment]
> Hey! Can I try for this ?

Can you share how you would be working on it?

**Comment by Harshil-Jani** [comment]
> > Hey! Can I try for this ?
> 
> Can you share how you would be working on it?

@prakhar728 It goes like, using a display `none` property to hide the element when not required. Keeping the Backend in mind, It was suitable to keep all on same page so I tried implementing this. And used Bootstrap for components. And basic event listners in JS. Any Suggestion if you got, Let me know ! ✌🏻

**Comment by prakhar728** [comment]
> > > Hey! Can I try for this ?
> > 
> > 
> > Can you share how you would be working on it?
> 
> @prakhar728 It goes like, using a display `none` property to hide the element when not required. Keeping the Backend in mind, It was suitable to keep all on same page so I tried implementing this. And used Bootstrap for components. And basic event listners in JS. Any Suggestion if you got, Let me know ! ✌🏻

Go ahead. Just keep in mind that the login route returns the JWT token.

**Comment by madrix01** [comment]
@Harshil-Jani We have used react js for frontend. You have made commit using vanilla javascript. Can you implement the same in react ?

**Comment by Harshil-Jani** [comment]
> @Harshil-Jani We have used react js for frontend. You have made commit using vanilla javascript. Can you implement the same in react ?

Actually I started React this week so I am not much familiar ! If we are not in hurry then by tomorrow let me go through few resources and I will let you know If I can do it with React or not !

**Comment by madrix01** [comment]
yea sure !! ask @prakhar728  for help.

**Comment by prakhar728** [comment]
> yea sure !! ask @prakhar728 for help.

Works with me @Harshil-Jani ping me anytime.

**Comment by Harshil-Jani** [comment]
> Go ahead. Just keep in mind that the login route returns the JWT token.

@prakhar728 I searched about this thing and I got few things, Please Correct me at places where I go wrong. 
We need to implement async and await functions in Javascript file and make sure the token is generated successfully. then we need to pass this token in login functions right? I think I need to learn more about JWT and React . Once Plz let me know any specific path that I can follow to fix both of these.

**Comment by prakhar728** [comment]
> > Go ahead. Just keep in mind that the login route returns the JWT token.
> 
> @prakhar728 I searched about this thing and I got few things, Please Correct me at places where I go wrong. We need to implement async and await functions in Javascript file and make sure the token is generated successfully. then we need to pass this token in login functions right? I think I need to learn more about JWT and React . Once Plz let me know any specific path that I can follow to fix both of these.

Yeah sure.  I have a repo under the name [Auth-template](https://github.com/prakhar728/auth_template). This has a React front end. You can view this or you can try and use it. (Note- This is not a promotion). If you need any help then ping me.

**Comment by Harshil-Jani** [comment]
Yo ! Guys I have gone through React docs and Little bit about JWT as well. Thanks @prakhar728 Your template provided somewhat idea how to work in React. And I think I can now give it a try.

**Comment by prakhar728** [comment]
> Yo ! Guys I have gone through React docs and Little bit about JWT as well. Thanks @prakhar728 Your template provided somewhat idea how to work in React. And I think I can now give it a try.

Sure go ahead!

**Comment by neutr0nStar** [comment]
Hey, I am comfortable with react, JWT, etc. May I start work on Login and registration. If yes, please assign it to me.
P.S. I have prior work experience in making such projects (internship)
